### PR TITLE
Change direct SSLSocket instantiation with wrap_socket

### PR DESCRIPTION
**Pull request by avborhanian**
Attempt at fixing the 3.7 issue.

### CODE REVIEW

**Comment by clokep** [comment]
Related to #206.

**Comment by clokep** [comment]
I tested this briefly in my testing environment which uses amqps via py-amqp on Python 2.7.6 on Ubuntu 14.04. I was able to connect OK to RabbitMQ with this change.

**Comment by thedrow** [comment]
I'll release this as soon as I can.

**Comment by auvipy** [comment]
If you document the release process, I can release more frequently hopefully

**Comment by dhananjaysathe** [comment]
This breaks on python 2.7 the wrap_context method returns out for some reason and does not execute any further code the wrapping does not happen and you end up with an invalid sni header. Do we have a test that actually tests for SNI ?

File: amqp/transport.py [hunk 1]
```diff
@@ -341,11 +341,16 @@ def _wrap_socket_sni(self, sock, keyfile=None, certfile=None,
...
         # Set SNI headers if supported
         if (server_hostname is not None) and (
-                hasattr(ssl, 'HAS_SNI') and ssl.HAS_SNI):
-            opts['server_hostname'] = server_hostname
-        sock = ssl.SSLSocket(**opts)
+                hasattr(ssl, 'HAS_SNI') and ssl.HAS_SNI) and (
+                hasattr(ssl, 'SSLContext')):
+            context = ssl.SSLContext(opts['ssl_version'])
+            context.verify_mode = cert_reqs
+            context.check_hostname = True
```
Comment (dhananjaysathe) [inline amqp/transport.py]: why is this line being set here ?

File: amqp/transport.py [hunk 2]
```diff
@@ -341,11 +341,16 @@ def _wrap_socket_sni(self, sock, keyfile=None, certfile=None,
...
+            context.load_cert_chain(certfile, keyfile)
```
Comment (dhananjaysathe) [inline amqp/transport.py]: also if this was none this should not be executed

Comment (thedrow) [reply amqp/transport.py]: Feel free to send a PR.

**Pull request merged**